prow/cmd/moonraker/main.go

```diff
@@ -0,0 +1,133 @@
+	port           int
+	cookiefilePath string
+
+	config configflagutil.ConfigOptions
+
+	dryRun                 bool
+	gracePeriod            time.Duration
+	instrumentationOptions prowflagutil.InstrumentationOptions
+}
+
+func gatherOptions(fs *flag.FlagSet, args ...string) options {
+	var o options
+	fs.IntVar(&o.port, "port", 8080, "HTTP port.")
+	fs.DurationVar(&o.gracePeriod, "grace-period", 180*time.Second, "On shutdown, try to handle remaining events for the specified duration. ")
+	fs.StringVar(&o.cookiefilePath, "cookiefile", "", "Path to git http.cookiefile, leave empty for github or anonymous")
+	for _, group := range []flagutil.OptionGroup{&o.client, &o.github, &o.instrumentationOptions, &o.config} {
+		group.AddFlags(fs)
+	}
+
+	fs.Parse(args)
+
+	return o
+}
+
+func (o *options) validate() error {
+	var errs []error
+	for _, group := range []flagutil.OptionGroup{&o.client, &o.github, &o.instrumentationOptions, &o.config} {
+		if err := group.Validate(o.dryRun); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func main() {
+	logrusutil.ComponentInit()
+
+	o := gatherOptions(flag.NewFlagSet(os.Args[0], flag.ExitOnError), os.Args[1:]...)
+	if err := o.validate(); err != nil {
+		logrus.WithError(err).Fatal("Invalid options")
+	}
+
+	configAgent, err := o.config.ConfigAgent()
+	if err != nil {
+		logrus.WithError(err).Fatal("Error starting config agent.")
+	}
+
+	// If we are provided credentials for Git hosts, use them. These credentials
+	// hold per-host information in them so it's safe to set them globally.
+	if o.cookiefilePath != "" {
+		cmd := exec.Command("git", "config", "--global", "http.cookiefile", o.cookiefilePath)
+		if err := cmd.Run(); err != nil {
+			logrus.WithError(err).Fatal("unable to set cookiefile")
+		}
+	}
+
+	gitClient, err := o.github.GitClientFactory(o.cookiefilePath, &o.config.InRepoConfigCacheDirBase, o.dryRun, false)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting Git client.")
+	}
+
+	cacheGetter, err := config.NewInRepoConfigCache(o.config.InRepoConfigCacheSize, configAgent, gitClient)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error creating InRepoConfigCacheGetter.")
+	}
+	metrics.ExposeMetrics("moonraker", configAgent.Config().PushGateway, o.instrumentationOptions.MetricsPort)
+
+	// Start serving liveness endpoint /healthz.
+	health := pjutil.NewHealthOnPort(o.instrumentationOptions.HealthPort)
+
+	mr := moonraker.Moonraker{
+		ConfigAgent:       configAgent,
+		InRepoConfigCache: cacheGetter,
+	}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/get-inrepoconfig", mr.ServeGetInrepoconfig)
+	server := &http.Server{
+		Addr:    ":" + strconv.Itoa(o.port),
+		Handler: mux,
+	}
+	logrus.Info("Listening on port 8080...")
+	interrupts.ListenAndServe(server, o.gracePeriod)
+	health.ServeReady(func() bool {
+		return true
+	})
+	interrupts.WaitForGracefulShutdown()
+	logrus.Info("Ended gracefully")
```

Comment:
nit: We already log this consistently here: https://github.com/listx/test-infra/blob/cfc412716ce3dc523d6903a5cad3af1ce8e42b6b/prow/interrupts/interrupts.go#L117

Reply:
Will update.

As a follow-up, we need to delete a similar line from Gangway.